tdrs-backend/tdpservice/search_indexes/documents.py

```diff
@@ -0,0 +1,375 @@
+"""Elasticsearch document mappings for Django models."""
+
+from django_elasticsearch_dsl import Document
+from django_elasticsearch_dsl.registries import registry
+from .models import T1, T2, T3, T4, T5, T6, T7
+
+
+@registry.register_document
+class T1DataSubmissionDocument(Document):
+    """Elastic search model mapping for a parsed T1 data file."""
+
+    class Index:
+        """ElasticSearch index generation settings."""
+
+        name = 't1_submissions'
+        settings = {
+            'number_of_shards': 1,
+            'number_of_replicas': 0,
+        }
+
+    class Django:
+        """Django model reference and field mapping."""
+
+        model = T1
+        fields = [
```

Comment:
Is there a DRY way to do this? I keep seeing these fields listed in multiple files and stretching file sizes.

Reply:
to my knowledge, no. DRF has an `__all__` modifier that allows you to shorthand-map all model fields to a serializer, but `django-elastic-drf` doesn't have anything like that for Documents. That said, we could do an `Object.keys` approach like you recommended, but frankly this is a pretty common pattern in Django/DRF apps anyway (example in a `serializers` file https://github.com/raft-tech/TANF-app/blob/develop/tdrs-backend/tdpservice/users/serializers.py, we "redefine" fields all over the place for model -> serializer mappings, we do a similar thing with `ModelAdmin` definitions in `admin.py` files). Ultimately it fails quickly (when the application starts, it will attempt to verify field mappings, if there's an issue it should fail immediately after startup) and "redefining" the fields allowing you to customize the mappings of some fields in the future.